File: AppChat/app/src/main/java/com/example/appchat/calling/ui/LoginActivity.java
//package com.example.appchat.calling.ui;
//
//import android.content.Intent;
//import android.os.Bundle;
//
//import androidx.appcompat.app.AppCompatActivity;
//
//import com.example.appchat.calling.repository.MainRepository;
//import com.example.appchat.databinding.ActivityLogin4Binding;
//import com.permissionx.guolindev.PermissionX;
//
//public class LoginActivity extends AppCompatActivity {
//
//    private ActivityLogin4Binding views;
//
//    private MainRepository mainRepository;
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        views = ActivityLogin4Binding.inflate(getLayoutInflater());
//        setContentView(views.getRoot());
//        init();
//    }
//
//    private void init() {
//        mainRepository = MainRepository.getInstance();
//        views.enterBtn.setOnClickListener(v -> {
//            PermissionX.init(this)
//                    .permissions(android.Manifest.permission.CAMERA, android.Manifest.permission.RECORD_AUDIO)
//                    .request((allGranted, grantedList, deniedList) -> {
//                        if (allGranted) {
//                            //login to firebase here
//
//                            mainRepository.login(
//                                    views.username.getText().toString(), getApplicationContext(), () -> {
//                                        //if success then we want to move to call activity
//                                        startActivity(new Intent(LoginActivity.this, CallActivity.class));
//                                    }
//                            );
//                        }
//                    });
//
//
//        });
//    }
//}

File: AppChat/app/src/main/java/com/example/appchat/videocall/ui/LoginActivity.kt
package com.example.appchat.videocall.ui

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.appchat.databinding.ActivityLogin2Binding
import com.example.appchat.videocall.repository.MainRepository
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class LoginActivity : AppCompatActivity() {

    private lateinit var  views: ActivityLogin2Binding
    @Inject
    lateinit var mainRepository: MainRepository

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        views = ActivityLogin2Binding.inflate(layoutInflater)
        setContentView(views.root)
        init()
    }


    private fun init(){
        views.apply {
            btn.setOnClickListener {
                mainRepository.login(
                    usernameEt.text.toString(),passwordEt.text.toString()
                ){ isDone, reason ->
                    if (!isDone){
                        Toast.makeText(this@LoginActivity, reason, Toast.LENGTH_SHORT).show()
                    }else{
                        //start moving to our main activity
                        startActivity(Intent(this@LoginActivity, MainActivity::class.java).apply {
                            putExtra("username",usernameEt.text.toString())
                        })
                    }
                }
            }
        }
    }
}